### PR TITLE
Add CodeStyle Rule for MaxLineLength

### LINKED ISSUE #56: Rule: Max line length
I would like to be able to enforce a maximum line length on the files being linted

### PR DESCRIPTION
**Pull request by Mauin**
Resolves #56 

This feels like a very basic, stupid implementation walking though each source file line by line.

Happy for suggestions on how to improve the Rule. Was at first trying to walk the PSI Tree, but that would require lots of manual whitespace and codeword counting if I understood it correctly.

### CODE REVIEW

File: detekt-cli/src/main/resources/default-detekt-config.yml [hunk 1]
```diff
@@ -96,6 +96,9 @@ style:
   active: true
   WildcardImport:
     active: true
+  MaxLineLength:
+    active: true
+    maxLineLength: 100
```
Comment (arturbosch) [inline detekt-cli/src/main/resources/default-detekt-config.yml]: Intellij defines 120 as default so we should also use 120.

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxLineLength.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import org.jetbrains.kotlin.psi.KtFile
+import java.io.BufferedReader
+import java.io.InputStreamReader
+
+class MaxLineLength(config: Config = Config.empty) : Rule("MaxLineLength", Severity.Style, config) {
+
+	private val maxLineLength: Int = withConfig { valueOrDefault(MAX_LINE_LENGTH) { "100" } }.toInt()
+
+	override fun visit(root: KtFile) {
+		val bufferedReader = BufferedReader(InputStreamReader(root.text.byteInputStream()))
```
Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxLineLength.kt]: Did you write that having performance in mind? I'm wondering if `root.text.splitToSequence("\n")` would be clearer. I did some `performance tests` on kotlin (500k LLOC) and detekt, both implementations are ~ same.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxLineLengthSpec.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package io.gitlab.arturbosch.detekt.rules.style
+
+import io.gitlab.arturbosch.detekt.api.Rule
+import io.gitlab.arturbosch.detekt.rules.Case
+import io.gitlab.arturbosch.detekt.rules.load
+import io.gitlab.arturbosch.detekt.test.TestConfig
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class MaxLineLengthSpec {
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxLineLengthSpec.kt]: Call it Test not Spec, your not using Spek :)

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxLineLengthSpec.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+	val root = load(Case.MaxLineLength)
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxLineLengthSpec.kt]: Do not use load(), i just forgot to remove it :D. We have `Rule.lint(..)` and `Rule.format(...)` as expension functions. 
Try something like this:
```kotlin
internal class MaxLineLengthTest : RuleTest {
	
	override val rule: Rule = MaxLineLength()

	@Test
	fun test() {
		val findings = rule.lint(Case.ComplexClass.path())
		assertThat(rule.findings).hasSize(expected)
	}
}
```

If you wonder about two times reading the file, you can use:

```kotlin
val ktFile = compileForTest(Case.ComplexClass.path()).text
val findings = rule.lint(ktFile)
```

File: detekt-cli/src/main/resources/default-detekt-config.yml (hunk 1 above)
Comment (Mauin) [inline detekt-cli/src/main/resources/default-detekt-config.yml]: Oh, I thought so as well but my IntelliJ was set to 100, I must've changed it at some point. I'll change it 👍

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxLineLengthSpec.kt (hunk 3 above)
Comment (Mauin) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxLineLengthSpec.kt]: I'll either rename, or change the test to use Spek 👍

**Comment by arturbosch** [comment]
Awesome thx!

**Pull request merged**